lib/dotcom/alerts/disruptions/subway.ex

```diff
@@ -46,6 +48,38 @@ defmodule Dotcom.Alerts.Disruptions.Subway do
     |> Enum.into(%{})
   end
 
+  # Given an alert with multiple active periods, some of which may be
+  # contiguous with one another (next one starts the same day or the
+  # day after the previous one), combines the contiguous active
+  # periods, and then returns
+  defp separate_active_periods(alert) do
+    alert.active_period
+    |> combine_contiguous_active_periods()
+    |> Enum.map(fn active_period ->
+      %Alerts.Alert{alert | active_period: [active_period]}
+    end)
+  end
+
+  # Given a list of active periods
+  defp combine_contiguous_active_periods([active_period1, active_period2 | rest]) do
+    {start1, end1} = active_period1
+    {start2, end2} = active_period2
+
+    if Timex.before?(
+         ServiceDateTime.service_date(end1) |> Timex.shift(days: 1),
```

Comment:
I think you meant to use the other `shift` here?

Reply:
Well, uh... yes I did 👀 

In this case, I've decided that I want `Date.shift/2`, which I had not known about until about 30 seconds ago.

Reply:
[PTALA](https://github.com/mbta/dotcom/pull/2402/commits/c5a9bcf4f52ee3cd6282a10df441054ef78adb89)